broker/src/main/java/io/camunda/zeebe/broker/system/partitions/impl/steps/ZeebeDbPartitionStep.java

```diff
@@ -54,4 +43,36 @@
   public String getName() {
     return "ZeebeDb";
   }
+
+  private void recoverDb(
+      final PartitionTransitionContext context, final CompletableActorFuture<Void> openFuture) {
+    final ActorFuture<Void> recoverFuture;
+    try {
+      recoverFuture = context.getStateController().recover();
```

Comment:
:x: since it returns a future, personally I would expect it would complete the future exceptionally instead of throwing an exception